
syntax = "proto3";

service SimInspector {
  rpc SendCommand(Command) returns (CommandResponse) {}
  rpc GetFramebuffer(FramebufferRequest) returns (stream FramebufferSegment) {}
}

enum CommandType {
  PAUSE = 0;
  RUN = 1;
  STEP = 2;
  STOP = 3;
}

message Command {
  CommandType command = 1;
  uint32 step_size = 2;  
}

message CommandResponse {
  bool ok = 1;
}

message FramebufferRequest {
  optional FramebufferRegion region = 1;
}
message FramebufferRegion {
  uint32 x = 1;
  uint32 y = 2;
  uint32 pix = 3;
}

message FramebufferSegment {
  repeated uint32 data = 1;
}

